proto/stream_plan.proto

```diff
@@ -435,8 +427,6 @@ message ArrangeNode {
   repeated uint32 distribution_key = 2;
   // Used for internal table states.
   catalog.Table table = 3;
-  // Used to control whether doing sanity check, open it when upstream executor is source executor.
-  HandleConflictBehavior handle_pk_conflict_behavior = 4;
```

Comment:
To not mark this PR as breaking change, we have to:
1. either keep this field and mark it as deprecated
2. or write a warning comment to reserve `4` (seems unreliable)

Reply:
> 2\. reserve `4`

This is possible with the Protobuf's `reserved`. 

https://protobuf.dev/programming-guides/proto3/#reserved

Reply:
Exactly what we want.